### ISSUE TITLE
What bot?

### ISSUE THREAD

**Issue by PBJI** [PBJI/ultimate-tic-tac-toe-2#6]
List of AI agents we could potentially use (Please note, the author of this issue had limited knowledge about the vast variety of AI solutions, hence the list above would not reflect the actual optimized solution yet)

1) Minimax
2) Heuristic based
3) Re-inforcement learning.

The complexity of each solution increase by the order.

**Comment by PBJI** [comment]
### UTTT - Agent
While tic-tac-toe is elementary to solve[5] and can be done nearly instantly using depth-first search, ultimate tic-tac-toe cannot be reasonably solved using any brute-force tactics. Therefore, more creative computer implementations are necessary to play this game.

The most common artificial intelligence (AI) tactic, minimax, may be used to play ultimate tic-tac-toe, but has difficulty playing this. This is because, despite having relatively simple rules, ultimate tic-tac-toe lacks any simple heuristic evaluation function. This function is necessary in minimax, for it determines how good a specific position is. Although elementary evaluation functions can be made for ultimate tic-tac-toe by taking into account the number of small board victories, these largely overlook positional advantage that is much harder to quantify. Without any efficient evaluation function, most typical computer implementations are weak, and therefore there are few computer opponents that can consistently outplay humans.

However, artificial intelligence algorithms that don't need evaluation functions, like the Monte Carlo tree-search algorithm, have no problem in playing this game. The Monte Carlo tree search relies on random simulations of games to determine how good a position is instead of a positional evaluation and is therefore able to accurately assess how good a current position is. Therefore, computer implementations using these algorithms tend to outperform minimax solutions and can consistently beat human opponents.

**Comment by PBJI** [comment]
[Monte-Carlo tree algorithm is better than Minimax](https://philippmuens.com/minimax-and-mcts) 

tl;dr

Monte Carlo is easy to implement in most games, even though Minimax function with alpha-beta pruning could be more accurate. The compromise on accuracy comes at saving of developing tome for heuristics, especially for games as complex as Ultimate TicTacToe.

**Comment by NitkarshChourasia** [comment]
Okay.
